### DISCUSSION TITLE
Ngrok option

### DISCUSSION THREAD

**Discussion by sitzbrau** [rsxdalv/TTS-WebUI#310] [Ideas]
Hi 
could you please add an "**--ngrok** _*authcode*_" option to server.py?
thanks

**Comment by rsxdalv** [comment]
Sure, I think Gradio supports it. How would you like to use it? Directly
from command line like server.py --ngrok authcode?

**Reply by sitzbrau** [reply]
yes, directly from command line it would be great because i'm trying to install it on an amazon ec2 instances with ubuntu cli

**Reply by rsxdalv** [reply]
Ah, my bad, gradio doesn't support it automatically, gradio only has the --share flag.
There is an article specifically for this webui + Amazon:

https://medium.com/@dminhk/how-to-run-tts-generation-webui-on-amazon-sagemaker-notebook-55bc39f4e14f

I would sort of prefer that the ngrok be separate due to security and separation of concerns. I.e., personally I'd probably use this as a container and have terraform (opentofu) + kubernetes handle the networking and proxying separately.

Please let me know how you see this, I don't use ngrok much so I might be off base.
